src/test/throttler/throttler.spec.ts

```diff
@@ -236,6 +243,152 @@ const throttlerTest = (throttlerConstructor: ThrottlerConstructor) => {
     expect(q.run(2)).to.eventually.to.equal("result: 2");
     expect(q.run(3)).to.eventually.to.equal("result: 3");
   });
+
+  it("should resolve if task finishes before timeout", async () => {
+    const handler = (task: number) => {
+      return Promise.resolve(`result: ${task}`);
+    };
+
+    const q = new Queue({
+      handler,
+    });
+
+    expect(await q.run(2, 20000000)).to.equal("result: 2");
+    expect(q.complete).to.equal(1);
+    expect(q.success).to.equal(1);
+    expect(q.errored).to.equal(0);
+    expect(q.retried).to.equal(0);
+    expect(q.total).to.equal(1);
+  });
+
+  it("should reject if timeout", async () => {
+    const handler = (task: number) =>
+      new Promise((resolve) => {
+        setTimeout(() => {
+          resolve(`result: ${task}`);
+        }, 150);
+      });
+
+    const q = new Queue({
+      handler,
+    });
+
+    try {
+      await q.run(2, 100);
```

Comment:
If this succeeds, the test would not fail. I think this is true of a number of these tests (but I won't comment on each).

If you like this pattern, I'd suggest doing something like:

```typescript
let err: error;
try {
  await...
} catch (e) {
  err = e;
}
expect(err).to.be...
```

You could also do:

```typescript
expect(async () => {
  await q.run(2, 100);
}).to.be.rejectedWith(TimeoutError, "Task index 0 failed...")
```

(I think that's close - it's untested)

Reply:
Moved error assertions out of try/catch block